docs/environment.md

```diff
@@ -17,3 +96,28 @@ The following environment variables are set by pixi, when using the `pixi run`,
 - `CONDA_PREFIX`: The path to the environment. (Used by multiple tools that already understand conda environments)
 - `CONDA_DEFAULT_ENV`: The name of the environment. (Used by multiple tools that already understand conda environments)
 - `PATH`: We prepend the `bin` directory of the environment to the `PATH` variable, so you can use the tools installed in the environment directly.
+
+!!! note
+    The variables are set by pixi but can not be overwritten, so you can not change the root of the project by setting `PIXI_PROJECT_ROOT` in the environment.
+
+## Solving environments
+When you run a command that uses the environment, pixi will check if the environment is in sync with the `pixi.lock` file.
+If it is not, pixi will solve the environment and update it.
+This means that pixi will retrieve the best set of packages for the dependency requirements that you specified in the `pixi.toml` and will put the output of the solve step into the `pixi.lock` file.
+Solving is a mathematical problem and can take some time, but we take pride in the way we solve environments, and we are confident that we can solve your environment in a reasonable time.
+If you want to learn more about the solving process, you can read these:
+
+- [Rattler(conda) resolver blog](https://prefix.dev/blog/the_new_rattler_resolver)
+- [Rip(PyPI) resolver blog](https://prefix.dev/blog/introducing_rip)
+
+Pixi solves both the `conda` and `PyPI` dependencies, where the `PyPI` dependencies use the conda packages as a base, so you can be sure that the packages are compatible with each other.
+These solvers are split between the `rattler` and `rip` library, these do the heavy lifting of the solving process.
```

Comment:
Maybe link to resolvo as the solver, that is used in both projects?

Reply:
Yeah good one, extended this a with some more info